contributors/devel/sig-testing/testing-strategy.md

```diff
@@ -0,0 +1,99 @@
+## Defining a Robust Testing Strategy
+
+This document outlines a testing strategy for Kubernetes features based on the **testing pyramid**, with considerations on the existing CI system's characteristics.
+
+### The Testing Pyramid
+
+Prioritize tests based on the testing pyramid, refer to the [Testing Guide](./testing.md):
+
+- **Unit Tests:**  The foundation. Fast, isolated, and cover individual components.
+- **Integration Tests:** Verify interactions between components within your subsystem.
```

Comment:
```suggestion
- **Integration Tests:** Verify interactions between components within your subsystem. These are preferred for tests which require cluster components to be run with test-specific configurations.
```

Reply:
I think this is a brief description of what is expected to test here, I still mention that is a recommended strategy so saying that is is not mandatory or is not covered by one or the other sounds like a bit redundant

Reply:
For completeness, the lack of integration tests and the replacing them with e2e test is a source of flakiness and unreliability on CI systems for large software  projects.

I've worked with the CI of two major kubernetes distros and other big OSS project like Openstack that have this problem in their testing strategy, e2e tests to cover integrationg is super unreliable and flaky ...  integrations give you the middle ground to avoid large exposure to environmental problems (like the network) and the confidence to identify the owners of the e2e test failures (that generally speaking is the problem of flakes, who owns and who debugs it , ... 🤷 ) ... but as everything , these are general guidelines

Reply:
Agreed. I brought this up because I've sometimes seen comments about "your feature must have integration tests, it's required" and I think *that* statement goes too far. Integration tests make sense... when they make sense :grin: 

Related to this, I think our helper packages for writing E2E tests are in a better shape than the corresponding helper packages for integration tests. Let's talk more about that at KubeCon, after https://kcsna2024.sched.com/event/1nSjo/unified-framework-for-unit-integration-and-e2e-testing